File: quark/test/emit/slackpack/js/slackpack.js
var _qrt = require("quark_runtime.js");
var pkg = require('./pkg');
exports.pkg = pkg;
var slack = require('./slack');
exports.slack = slack;




function main() {
    var cli = new slack.Client(null, "fake-token", new pkg.Handler());
    (cli).onWSMessage(null, "{\"type\": \"hello\"}");
    (cli).onWSMessage(null, "{\"type\": \"message\", \"user\": \"uid-1\", \"channel\": \"chanel-1\"}");
}
exports.main = main;

main();

File: quarkc/test/emit/expected/rb/slackpack/lib/slackpack.rb
module Quark
require "quark"
def self.slackpack; Slackpack; end
module Slackpack
require_relative 'slack' # 0 () ()
require_relative 'pkg' # 0 () ()
require_relative 'slackpack' # 0 () ()

def self.call_main()
    self.main(::DatawireQuarkCore::List.new(ARGV))

    nil
end
def self.main(args)
    
    cli = ::Quark.slack.Client.new(nil, "fake-token", ::Quark.pkg.Handler.new())
    cli.onWSMessage(nil, "{\"type\": \"hello\"}")
    cli.onWSMessage(nil, "{\"type\": \"message\", \"user\": \"uid-1\", \"channel\": \"chanel-1\"}")


    nil
end

if __FILE__ == $0 then ::Quark.slackpack.call_main() end

end # module Slackpack
end # module Quark
